ldaptive/src/main/java/org/ccci/idm/user/ldaptive/dao/io/GroupValueTranscoder.java

```diff
@@ -1,96 +1,68 @@
 package org.ccci.idm.user.ldaptive.dao.io;
 
-import com.google.common.base.Strings;
-import com.google.common.collect.ImmutableList;
-import com.google.common.collect.Lists;
+import com.google.common.annotations.VisibleForTesting;
 import org.ccci.idm.user.Dn;
 import org.ccci.idm.user.Group;
-import org.ldaptive.DnParser;
-import org.ldaptive.LdapAttribute;
+import org.ccci.idm.user.ldaptive.dao.util.DnUtils;
 import org.ldaptive.io.AbstractStringValueTranscoder;
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
-import java.util.List;
 
 public class GroupValueTranscoder extends AbstractStringValueTranscoder<Group> {
-    private static final String delimiter = ",";
-    private static final String valueDelimiter = "=";
+    @Nonnull
+    private Dn baseDn = Dn.ROOT;
 
-    private String baseDn = "";
-
-    public String getBaseDn() {
-        return this.baseDn;
-    }
-
-    public void setBaseDn(@Nullable final String dn) {
-        this.baseDn = Strings.nullToEmpty(dn);
+    @Nonnull
+    @VisibleForTesting
+    String getBaseDnString() {
+        return DnUtils.toString(baseDn);
     }
 
-    @Deprecated
-    public String getPathRdnAttr() {
-        return "ou";
+    public void setBaseDn(@Nullable final Dn dn) {
+        baseDn = dn != null ? dn : Dn.ROOT;
     }
 
-    @Deprecated
-    public void setPathRdnAttr(@Nullable final String rdnAttr) {}
-
-    @Deprecated
-    public String getNameRdnAttr() {
-        return "cn";
+    /**
+     * Set the baseDn as a String. Provided for Spring ease of use.
+     *
+     * @param dn
+     */
+    public void setBaseDnString(@Nullable final String dn) {
+        setBaseDn(DnUtils.toDn(dn));
     }
 
-    @Deprecated
-    public void setNameRdnAttr(@Nonnull final String rdnAttr) {}
-
     @Override
     public Class<Group> getType() {
         return Group.class;
     }
 
     @Override
     public String encodeStringValue(@Nonnull final Group group) {
-        final StringBuilder sb = new StringBuilder();
-
-        // append components
-        for (final Dn.Component component : Lists.reverse(group.getComponents())) {
-            if(sb.length() > 0) {
-                sb.append(delimiter);
-            }
-            sb.append(component.type).append(valueDelimiter).append(LdapAttribute.escapeValue(component.value));
-        }
-
-        if (baseDn.length() > 0) {
-            sb.append(delimiter).append(this.baseDn);
+        if (!group.isDescendantOf(baseDn)) {
```

Comment:
This will throw an exception if your `baseDn` is equal to the `group` (same for `decodeStringValue`). Not sure if there is a use case for them being equal.